nestkernel/connection_manager.cpp

```diff
@@ -195,38 +201,40 @@ nest::ConnectionManager::set_synapse_status( index gid,
 void
 nest::ConnectionManager::delete_connections_()
 {
-  for ( tVSConnector::iterator it = connections_.begin();
-        it != connections_.end();
-        ++it )
+#ifdef _OPENMP
+#pragma omp parallel
   {
-    for ( tSConnector::nonempty_iterator iit = it->nonempty_begin();
-          iit != it->nonempty_end();
-          ++iit )
+#pragma omp for schedule( static, 1 )
```

Comment:
My hope was, that this makes the code clearer: there are already a lot `#ifdef`'s and not using the `#pragma omp for schedule( static, 1 )` would increase this further. Another point is, that the indentation becomes 'more' wrong, if we are compiling with OpenMP, as the outer loop is gone and the inner loop is indented one level to much. Here is a version with the `pragma omp for` removed:

```C++
void
nest::ConnectionManager::delete_connections_()
{
#ifdef _OPENMP
#pragma omp parallel
  {
    size_t t = kernel().vp_manager.get_thread_id();
#else
    for ( size_t t = 0; t < connections_.size(); ++t )
    {
#endif
      for (
        tSConnector::nonempty_iterator iit = connections_[ t ].nonempty_begin();
        iit != connections_[ t ].nonempty_end();
        ++iit )
      {
#ifdef USE_PMA
        validate_pointer( *iit )->~ConnectorBase();
#else
      delete validate_pointer( *iit );
#endif
      }
      connections_[ t ].clear();
#ifdef _OPENMP
    }
#endif

#if defined _OPENMP && defined USE_PMA
#ifdef IS_K
    poormansallocpool[ t ].destruct();
    poormansallocpool[ t ].init();
#else
    poormansallocpool.destruct();
    poormansallocpool.init();
#endif
#endif

#ifdef _OPENMP
  }
#endif
}
```